services/customdestinationmanager/customdestinationmanager.go

```diff
@@ -2,31 +2,213 @@ package customdestinationmanager
-func Send(jsonData json.RawMessage, destination string) (int, string, string) {
+func Send(jsonData json.RawMessage, destination string, producer interface{}, config interface{}) (int, string, string) {
 
 	switch {
 	case misc.ContainsString(objectStreamDestinations, destination):
-		return streammanager.Produce(jsonData, destination)
+		return streammanager.Produce(jsonData, destination, producer, config)
 	default:
 		return 404, "No provider configured for StreamManager", ""
 	}
 
 }
+
+// SendData gets the producer from destinationConfigProducerMap and sends data
+func (customManager *CustomManagerT) SendData(jsonData json.RawMessage, sourceID string, destID string) (int, string, string) {
+
+	var respStatusCode int
+	var respStatus, respBody string
+
+	destination := customManager.destination
+	key := sourceID + "-" + destID
+
+	producerConfig := ProducerConfig{}
+
+	producerCreationLock.RLock()
+	if destinationConfigProducerMap[key] != (ProducerConfig{}) {
+
+		producerConfigFromMap := destinationConfigProducerMap[key]
+		producer := producerConfigFromMap.Producer
+		config := producerConfigFromMap.Config
+		producerCreationLock.RUnlock()
+		if producer != nil {
+			producerConfig = producerConfigFromMap
+		} else {
+			/* As the producers are created when the server gets up or workspace config changes, it may happen that initially
+			the destinatin, such as Kafka server was not up, so, producer could not be created. But, after sometime it becomes reachable,
+			so, while sending the event if the producer is not creatd alrady, then it tries to create it. */
+			producerCreationLock.Lock()
+			producerConfigFromMap := destinationConfigProducerMap[key]
+			producer = producerConfigFromMap.Producer
+			config = producerConfigFromMap.Config
+			if producer != nil {
+				producerConfig = producerConfigFromMap
+			} else {
+				producer, err := GetProducer(config, destination)
+				producerConfig = ProducerConfig{Config: config, Producer: producer}
+				destinationConfigProducerMap[key] = producerConfig
+				if err != nil {
+					logger.Errorf("[%s Destination manager] error while creating producer for destination: %s with error %v ", customManager.destination, destID, err)
+				} else {
+					logger.Infof("[%s Destination manager] created new producer: %v for destination: %s and source %s", customManager.destination, producer, destID, sourceID)
+				}
+			}
+			producerCreationLock.Unlock()
+		}
+	} else {
+		producerCreationLock.RUnlock()
+	}
+
+	if producerConfig != (ProducerConfig{}) || producerConfig.Producer != nil {
+		respStatusCode, respStatus, respBody = Send(jsonData, destination, producerConfig.Producer, producerConfig.Config)
+	} else {
+		respStatusCode, respStatus, respBody = 400, "Producer not found in router", "Producer could not be created"
+	}
+
+	return respStatusCode, respStatus, respBody
+}
+
+// CreateOrUpdateProducer creates or updates producer based on destination config
```

Comment:
This function should return a producer and error. 
Then update  `destinationConfigProducerMap[key]` in the caller

Reply:
It is basically a sub part of backendConfigSubscriber and is called from that function only to update the config-producer map. For creating a new producer, there is a different method, named, newProducer(). Updated the comment. Does it make sense?